src/infra/ConversionTrackingReporter.ts

```diff
@@ -0,0 +1,75 @@
+import { ConversionEvent, COOKIE_PARAM, ListingsClickEvent } from '../models';
+import { DEFAULT_CONVERSION_TRACKING_DOMAIN, LISTINGS_SOURCE_PARAM } from '../models/constants';
+import { CommonConversionData } from '../models/conversiontracking/CommonConversionData';
+import { ConversionTrackingService, HttpRequesterService } from '../services';
+import { calculateSeed } from './CalculateSeed';
+
+const conversionEndpoint = 'conversiontracking/conversion';
+const listingsEndpoint = 'listings';
+
+export class ConversionTrackingReporter implements ConversionTrackingService {
+  private _debug: boolean|undefined;
+  constructor(
+    private httpRequesterService: HttpRequesterService,
+    private debug?: boolean|undefined,
+  ) {
+    this._debug = debug;
+  }
+
+  /**
+   * Prints event details to the console for debugging of analytics events as they fire.
+   * @param event - the details of the event that will be printed
+   * @param type - the type of the event that was tracked
+   */
+  private printEvent(event: string, type: string): void {
+    if (!this._debug) return;
+    console.log(
+      `%c[YextAnalytics]%c- Tracked ${type} event: ${event}`,
+      'background: white; color: blue;',
+      '',
+    );
+  }
+
+  async handleRequest(url: string): Promise<void> {
+    const res = await this.httpRequesterService.get(url.toString());
+    if (res.status !== 200 && !(res.type === 'opaqueredirect' || res.type === 'opaque')) {
+      const errorMessage = await res.text();
+      throw new Error(errorMessage);
+    }
+  }
+
+  private static formatBaseEvent(event: CommonConversionData, params: URLSearchParams): void {
+    params.set(COOKIE_PARAM, event.cookieId);
+    if (event.referrer) params.set('referrer', event.referrer);
+    params.set('v', calculateSeed().toString());
+  }
+
+  /** {@inheritDoc ConversionTrackingService.trackConversion} */
+  async trackConversion(event: ConversionEvent): Promise<void> {
+    const url = new URL(`https://${DEFAULT_CONVERSION_TRACKING_DOMAIN}/${conversionEndpoint}`);
+    const params = new URLSearchParams();
+    params.set('cid', event.cid);
+    if (event.cv) params.set('cv', event.cv);
+    ConversionTrackingReporter.formatBaseEvent(event, params);
+    url.search = params.toString();
+    await this.handleRequest(url.toString());
+    this.printEvent(event.cid, 'Conversion');
+  }
+
+  /** {@inheritDoc ConversionTrackingService.trackListings} */
+  async trackListings(event: ListingsClickEvent): Promise<void> {
```

Comment:
Curious, how is this expected to be used? I'm reading through the ytag code and the listings stuff has some special logic where it only fires once upon load.

Reply:
So some of the borders of this library are a bit weird because it's meant to be useable server side.  I need to think about the pageView method and what it should provide implementers to tell them to remove the listings parameter from the pageurl if necessary.  Any thoughts?  I can pull the listings crediting out of the pageView method so that it's entirely on the implementer to do and add that logic into `yext/pages` (seems like that might be best).  I could also add something to utils for implementers to use to manage that state

Reply:
Oh i see the code in PagesAnalyticsReporter where it does the only fire once logic. I think we might actually be ok with what logic already exists there, it seems that on the ytag implementation refreshing the page would cause the listings event to fire again. 

Up to you if you want to move it out right now, think it should be ok.

Reply:
oh cool I was just looking at ytag and hadn't seen your comment and yeah I agree it doesn't seem to handle page refreshes.  I can add a TODO to handle that in the future